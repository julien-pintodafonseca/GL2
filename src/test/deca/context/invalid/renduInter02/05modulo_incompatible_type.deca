// Description:
//      Opérateur modulo incorrect au niveau du typage
//      => l'opérateur modulo est uniquement applicable sur un entier
//
// Resultats:
//      Erreur contextuelle
//      Ligne 10 : Type float incompatible avec l'opérateur modulo

{
    float x = 7.2 % 1;
    println("x=", x);
}