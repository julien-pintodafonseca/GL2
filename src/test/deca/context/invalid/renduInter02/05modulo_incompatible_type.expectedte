src\test\deca\context\invalid\renduInter02\05modulo_incompatible_type.deca:10:18: L'opérateur % n'est utilisable qu'avec deux expressions de type int. Or, il est actuellement utilisé avec des expressions de type : float (pour 0x1.ccccccp2) et int (pour 1).
