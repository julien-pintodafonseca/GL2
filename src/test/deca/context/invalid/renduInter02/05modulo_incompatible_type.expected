05modulo_incompatible_type.deca:10:18: L'opérateur % n'est utilisable qu'avec deux expressions de type int. Or, il est actuellement utilisé avec des expressions de type : float (pour 0x1.ccccccp2) et int (pour 1).
